lib/rules/prefer-regex-literals.js

```diff
@@ -306,10 +354,61 @@ module.exports = {
 
                 for (const { node } of tracker.iterateGlobalReferences(traceMap)) {
                     if (disallowRedundantWrapping && isUnnecessarilyWrappedRegexLiteral(node)) {
+                        const regexNode = node.arguments[0];
+
                         if (node.arguments.length === 2) {
-                            context.report({ node, messageId: "unexpectedRedundantRegExpWithFlags" });
+                            const outputs = [];
+
+                            if (sourceCode.getCommentsInside(node).length <= 0) {
```

Comment:
I don't think this can be less than zero. :)

```suggestion
                            if (sourceCode.getCommentsInside(node).length === 0) {
```